apm-commons/apm-trace/src/main/java/org/skywalking/apm/trace/Span.java

```diff
@@ -293,18 +293,29 @@ public Span log(Throwable t) {
     private enum ThrowableTransformer {
         INSTANCE;
 
+        private static final int MAX_STACK_ELEMENT_LENGTH = 120;
+
         private String convert2String(Throwable e, int maxLength) {
             ByteArrayOutputStream buf = null;
             StringBuilder expMessage = new StringBuilder();
             try {
                 buf = new ByteArrayOutputStream();
                 Throwable causeException = e;
+                int currentAppendStackLength = 0;
                 while (expMessage.length() < maxLength && causeException != null) {
+                    int causeExceptionStackLength = causeException.getStackTrace().length;
+                    if (currentAppendStackLength + causeExceptionStackLength > MAX_STACK_ELEMENT_LENGTH) {
+                        break;
+                    }
+
                     causeException.printStackTrace(new java.io.PrintWriter(buf, true));
                     expMessage.append(buf.toString());
                     causeException = causeException.getCause();
+                    currentAppendStackLength += causeExceptionStackLength;
                 }
 
+            } catch (Throwable ee) {
+                expMessage.append("Failed to execute org.skywalking.apm.trace.ThrowableTransformer#convert2String, Exception message : " + ee.getMessage());
```

Comment:
This is safe mechanism, and the message will be at user UI. So I don't like `Failed to execute org.skywalking.apm.trace.ThrowableTransformer#convert2String,`, please remove them.

If you want add some comments, you can add these `Stack isn't avaible, message is:`